src/dataframe/dataframe.jl

```diff
@@ -768,21 +782,21 @@ function insertcols!(df::DataFrame, col_ind::Int; makeunique::Bool=false, name_c
 end
 
 
-##############################################################################
-##
-## Copying
-##
-##############################################################################
-
-# A copy of a DataFrame points to the original column vectors but
-#   gets its own Index.
-Base.copy(df::DataFrame) = DataFrame(copy(_columns(df)), copy(index(df)))
+"""
+    copy(df::DataFrame; copycolumns::Bool=true)
 
-##############################################################################
-##
-## Deletion / Subsetting
-##
-##############################################################################
+    Copy a `DataFrame` `df`.
+    By default, when `copycolumns` keyword argument is `true`,
+    all columns contained in the `DataFrame` are copied.
+    When `copycolumns` is `false` the columns from `df` are reused.
+"""
+function Base.copy(df::DataFrame; copycolumns::Bool=true)
+    if copycolumns
+        df[:, :]
```

Comment:
Is this more efficient than just calling the same constructor as below with `copycolumns=copycolumns`?

Reply:
we do one conversion less.